### PR TITLE
Use input facade in simple strategies as a custom fs

### PR DESCRIPTION
**Pull request by SparshithNR**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,7 +40,7 @@
   "dependencies": {
     "broccoli-debug": "^0.6.5",
     "broccoli-kitchen-sink-helpers": "^0.3.1",
-    "broccoli-plugin": "^3.1.0",
+    "broccoli-plugin": "^4.0.1",
     "ensure-posix-path": "^1.0.2",
     "fast-sourcemap-concat": "^1.4.0",
```
Comment (stefanpenner) [inline package.json]: Do we not need to bump this to 2.1.0?

File: concat.js [hunk 2]
```diff
@@ -109,15 +109,17 @@ module.exports = class Concat extends Plugin {
...
         separator: this.separator,
         header: this.header,
         headerFiles: this.headerFiles,
         footerFiles: this.footerFiles,
         footer: this.footer,
         contentLimit: this.contentLimit,
         baseDir: this.inputPaths[0]
-      }));
+      });
+      options['fs'] = this.input;
```
Comment (stefanpenner) [inline concat.js]: why a computed property`['fs']` over a regular property `options.fs = this.input;`

File: package.json (hunk 1 above)
Comment (SparshithNR) [reply package.json]: Yes. We have to I did it locally didn't push that change. My bad. Updated it

File: concat.js (hunk 2 above)
Comment (SparshithNR) [reply concat.js]: I am not sure why I did it. I have changed it to `option.fs`

**Comment by stefanpenner** [comment]
released as v4.2.2 🎉

**Pull request merged**